legistar/events.py

```diff
@@ -138,6 +138,29 @@ def ical(self, ical_text):
 
 
 class LegistarAPIEventScraperBase(LegistarAPIScraper):
+    webscraper_class = LegistarScraper
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._webscraper = self._init_webscraper()
+
+    def _init_webscraper(self):
+        webscraper = self.webscraper_class(
+            requests_per_minute=self.requests_per_minute,
+            retry_attempts=0)
+
+        if self.cache_storage:
+            webscraper.cache_storage = self.cache_storage
+
+        if self.requests_per_minute == 0:
+            webscraper.cache_write_only = False
+
+        webscraper.BASE_URL = self.WEB_URL
+
+        return webscraper
+
+    def _get_web_event(self, api_event):
```

Comment:
nice error! Might also be an opportunity to an AbstractBaseClass? https://stackoverflow.com/questions/3570796/why-use-abstract-base-classes-in-python

Up to you.

Reply:
@hancush ^

Reply:
Oh boy, I like that! The long-term maintainability argument [here](https://dbader.org/blog/abstract-base-classes-in-python) was very compelling. Switched!